docs/modules/ROOT/pages/api/governance.adoc

```diff
@@ -1982,6 +1990,560 @@ Emits a {TimelockUpdated} event.
+
+Updates the quorum value to `new_quorum`.
+
+[.contract-item]
+[[MultisigComponent-submit_transaction]]
+==== `[.contract-item-name]#++submit_transaction++#++(ref self: ContractState, to: ContractAddress, selector: felt252, calldata: Span<felt252>, salt: felt252)++` [.item-kind]#external#
+
+Submits a new transaction for confirmation.
+
+[.contract-item]
+[[MultisigComponent-submit_transaction_batch]]
+==== `[.contract-item-name]#++submit_transaction_batch++#++(ref self: ContractState, calls: Span<Call>, salt: felt252)++` [.item-kind]#external#
+
+Submits a new batch transaction for confirmation.
+
+[.contract-item]
+[[MultisigComponent-confirm_transaction]]
+==== `[.contract-item-name]#++confirm_transaction++#++(ref self: ContractState, id: TransactionID)++` [.item-kind]#external#
+
+Confirms a transaction with the given `id`.
+
+[.contract-item]
+[[MultisigComponent-revoke_confirmation]]
+==== `[.contract-item-name]#++revoke_confirmation++#++(ref self: ContractState, id: TransactionID)++` [.item-kind]#external#
+
+Revokes a previous confirmation for a transaction with the given `id`.
+
+[.contract-item]
+[[MultisigComponent-execute_transaction]]
+==== `[.contract-item-name]#++execute_transaction++#++(ref self: ContractState, to: ContractAddress, selector: felt252, calldata: Span<felt252>, salt: felt252)++` [.item-kind]#external#
+
+Executes a confirmed transaction.
+
+[.contract-item]
+[[MultisigComponent-execute_transaction_batch]]
+==== `[.contract-item-name]#++execute_transaction_batch++#++(ref self: ContractState, calls: Span<Call>, salt: felt252)++` [.item-kind]#external#
+
+Executes a confirmed batch transaction.
+
+[.contract-item]
+[[MultisigComponent-hash_transaction]]
+==== `[.contract-item-name]#++hash_transaction++#++(self: @ContractState, to: ContractAddress, selector: felt252, calldata: Span<felt252>, salt: felt252)++` [.item-kind]#external#
+
+Returns the computed identifier of a transaction containing a single call.
+
+[.contract-item]
+[[MultisigComponent-hash_transaction_batch]]
+==== `[.contract-item-name]#++hash_transaction_batch++#++(self: @ContractState, calls: Span<Call>, salt: felt252)++` [.item-kind]#external#
+
+Returns the computed identifier of a transaction containing a batch of calls.
+
+[#MultisigComponent-Internal-Functions]
+==== Internal functions
+
+[.contract-item]
+[[MultisigComponent-initializer]]
+==== `[.contract-item-name]#++initializer++#++(ref self: ContractState, quorum: u32, signers: Span<ContractAddress>)++` [.item-kind]#internal#
+
+Initializes the Multisig component with the initial `quorum` and `signers`.
+
+Emits a `SignerAdded` event for each signer added.  
+Emits a `QuorumUpdated` event if the quorum changes.
+
+[.contract-item]
+[[MultisigComponent-resolve_tx_state]]
+==== `[.contract-item-name]#++resolve_tx_state++#++(self: @ContractState, id: TransactionID) → TransactionState++` [.item-kind]#internal#
+
+Resolves and returns the current state of the transaction with the given `id`.
+
```

Comment:
I think it'd be helpful to include the possible states as they're listed in the in-code doc